torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -129,30 +130,90 @@ class TensorExprFuser {
   TensorExprFuser(std::shared_ptr<Graph> graph, size_t min_group_size)
       : graph_(std::move(graph)), min_group_size_(min_group_size) {}
 
+  // TODO: handle better
+  bool profiledWithDifferentTypes(Value* v) {
+    std::vector<TypePtr> types;
+    for (const auto& use : v->uses()) {
+      if (use.user->kind() == prim::profile) {
+        types.push_back(use.user->ty(attr::profiled_type));
+      }
+    }
+    for (size_t i = 1; i < types.size(); ++i) {
+      if (types.at(i - 1) != types.at(i)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void removeProfileNodesAndSpecializeTypes(Block* b) {
+    for (auto it = b->nodes().begin(); it != b->nodes().end(); it++) {
+      if (it->kind() == prim::profile) {
+        GRAPH_DEBUG("TYPEINFO ERASE: %", it->output()->debugName());
+        it->output()->replaceAllUsesWith(it->input());
+        if (!profiledWithDifferentTypes(it->input())) {
```

Comment:
Nit: maybe add the else statement with `GRAPH_DEBUG` to make this situation noticeable?